tests/utils_test/uncertainty_metrics_test.py

```diff
@@ -0,0 +1,241 @@
+            obs=env_state.obs,
+            next_obs=next_state.obs,
+            rewards=next_state.reward,
+            dones=next_state.done,
+            actions=actions,
+            truncations=next_state.info["truncation"],
+            state_desc=state_desc,
+            next_state_desc=next_state.info["state_descriptor"],
+        )
+
+        return next_state, policy_params, random_key, transition
+
+    # Create the initial environment states for samples and final indivs
+    reset_fn = jax.jit(jax.vmap(env.reset))
+    random_key, subkey = jax.random.split(random_key)
+    keys = jnp.repeat(jnp.expand_dims(subkey, axis=0), repeats=batch_size, axis=0)
+    init_states = reset_fn(keys)
+
+    # Create the scoring function
+    bd_extraction_fn = environments.behavior_descriptor_extractor[env_name]
+    scoring_fn = functools.partial(
+        scoring_function_brax_envs,
+        init_states=init_states,
+        episode_length=episode_length,
+        play_step_fn=play_step_fn,
+        behavior_descriptor_extractor=bd_extraction_fn,
+    )
+
+    # Evaluate in the Brax environment
+    fitnesses, descriptors, extra_scores, random_key = scoring_fn(
+        init_policies, random_key
+    )
+
+    # Initialise a container
+    min_bd, max_bd = env.behavior_descriptor_limits
+    centroids, random_key = compute_cvt_centroids(
+        num_descriptors=env.behavior_descriptor_length,
+        num_init_cvt_samples=num_init_cvt_samples,
+        num_centroids=num_centroids,
+        minval=min_bd,
+        maxval=max_bd,
+        random_key=random_key,
+    )
+    repertoire = MapElitesRepertoire.init(
+        genotypes=init_policies,
+        fitnesses=fitnesses,
+        descriptors=descriptors,
+        centroids=centroids,
+        extra_scores=extra_scores,
+    )
+
+    # Initialise an empty container for corrected repertoire
+    fitnesses = jnp.full_like(fitnesses, -jnp.inf)
+    empty_corrected_repertoire = MapElitesRepertoire.init(
+        genotypes=init_policies,
+        fitnesses=fitnesses,
+        descriptors=descriptors,
+        centroids=centroids,
+        extra_scores=extra_scores,
+    )
+
+    # Test that reevaluation_function runs and keeps at least one solution
+    keys = jnp.repeat(
+        jnp.expand_dims(subkey, axis=0), repeats=num_centroids, axis=0
+    )
+    init_states = reset_fn(keys)
+    reeval_scoring_fn = functools.partial(
+        scoring_function_brax_envs,
+        init_states=init_states,
+        episode_length=episode_length,
+        play_step_fn=play_step_fn,
+        behavior_descriptor_extractor=bd_extraction_fn,
+    )
+    corrected_repertoire, random_key = reevaluation_function(
+        repertoire=repertoire,
+        empty_corrected_repertoire=empty_corrected_repertoire,
+        scoring_fn=reeval_scoring_fn,
+        num_reevals=num_reevals,
+        random_key=random_key,
+    )
+    pytest.assume(jnp.any(fit_reproducibility_repertoire.fitnesses > -jnp.inf))
```

Comment:
Are we obliged to use a Brax environment here? (these make the pytest pipeline very slow)

Reply:
Good point, I replaced it with a noisy_arm environment.